File: ClassLibraryCisepro/DIVISION GEOGRÁFICA/ClassCiudades.vb
Imports System.Windows.Forms
Imports System.Data 
Imports ClassLibraryCisepro.ENUMS
Imports ClassLibraryCisepro.ProcesosSql

Namespace DIVISION_GEOGRÁFICA
    Public Class ClassCiudades
         
        Public Function BuscarIdCiudadPorNombreCiudad(ByVal tipoCon As TipoConexion, ByVal nombreCiudad As String) As Integer
            Dim pars = New List(Of Object())
            pars.Add(New Object() {"NOMBRE_CIUDADES", SqlDbType.NVarChar, nombreCiudad})
            Dim data = ComandosSql.SeleccionarQueryWithParamsToDataTable(tipoCon, "buscarIdCiudadPorNombreCiudad", True, pars) 
            Return If(data.Rows.Count = 0, 0, If(IsDBNull(data.Rows(0)(0)), 0, CInt(data.Rows(0)(0))))
        End Function

        Public Function BuscarNombreCiudadXIdCiudad(ByVal tipoCon As TipoConexion, ByVal idDeCiudad As Integer) As String
            Dim pars = New List(Of Object())
            pars.Add(New Object() {"ID_CIUDAD", SqlDbType.Int, idDeCiudad})
            Dim data = ComandosSql.SeleccionarQueryWithParamsToDataTable(tipoCon, "buscarNombreCiudadXIdCiudad", True, pars) 
            Return If(data.Rows.Count = 0, String.Empty, If(IsDBNull(data.Rows(0)(0)), String.Empty, CStr(data.Rows(0)(0))))
        End Function

        Public Function BuscarNombreCiudadXIdProvincia(ByVal tipoCon As TipoConexion, ByVal idProvincias As Integer) As DataTable
            Dim pars = New List(Of Object())
            pars.Add(New Object() {"ID_PROVINCIAS", SqlDbType.Int, idProvincias})
            Return ComandosSql.SeleccionarQueryWithParamsToDataTable(tipoCon, "BuscarNombreCiudadesXIdProvincia", True, pars) 
        End Function
         
        Private Function BuscarNombreCiudades(ByVal tipoCon As TipoConexion) As DataTable
            Return ComandosSql.SeleccionarQueryToDataTable(tipoCon, "buscarNombreCiudades", True) 
        End Function

        Public Function Autocompletar(ByVal tipoCon As TipoConexion) As AutoCompleteStringCollection
            Dim dt As DataTable = BuscarNombreCiudades(tipoCon)
            Dim coleccion As New AutoCompleteStringCollection()
            For Each row As DataRow In dt.Rows
                coleccion.Add(Convert.ToString(row("NOMBRE_CIUDADES")))
            Next
            Return coleccion
        End Function
 
    End Class
End Namespace

File: ClassLibrarySecurity/DivisionGeografica/ClassCiudades.cs
using System;
using System.Collections.Generic;
using System.Data;
using System.Data.SqlClient;
using ClassLibraryCisepro3.Enums;
using ClassLibraryCisepro3.ProcesosSql;

namespace ClassLibraryCisepro3.DivisionGeografica
{
    public class ClassCiudades
    {
        /**
         * ATRIBUTOS QUE ESTAN LIGADOS A LA(S) TABLA(S) CORRESPONDIENTE(S) 
         * SI AGREGO UNA COLUMNA EN LA TABLA (BD), ES RECOMENDABLE CREAR SU ATRIBUTO CORRESPONDIENTE CON EL MISMO TIPO DE DATO
         * TODOS LOS METODOS DE GUARDAR, SELECCIONAR, BUSCAR, ESTAN LIGADOS A UN PROCEDIMIENTO EN BD Y SU TIPO DE RETRNO GENERAL ES DATATABLE
         * HAY METODOS DONDE NO USAMOS PROCEDIMIENTOS ALMACENADOS SINO LAS SENTENCIAS SQL DE ACUERDO A LA LOGICA DE USO
         **/
        public int IdCiudadCiudades { get; set; }
        public string NombreCiudadesCiudades { get; set; }
        public int IdProvinciaCiudades { get; set; }
        public int EstadoCiudades { get; set; }


        public int BuscarMayorIdCiudades(TipoConexion tipoCon)
        {
            var data = ComandosSql.SeleccionarQueryToDataTable(tipoCon,"buscarMayorIdCiudades", true);
            return data.Rows.Count == 0 ? 0 : data.Rows[0][0] == DBNull.Value ? 0 : Convert.ToInt32(data.Rows[0][0]);
        }

        public DataTable BuscarNombreCiudadesXIdProvincia(TipoConexion tipoCon, string idp)
        {
            var pars = new List<object[]>
            {
                new object[] { "ID_PROVINCIAS", SqlDbType.Int, idp } 
            };
            return ComandosSql.SeleccionarQueryWithParamsToDataTable(tipoCon,"BuscarNombreCiudadesXIdProvincia", true, pars);
        } 

        public SqlCommand NuevoCiudad()
        {
            var cmd = new SqlCommand
            {
                CommandType = CommandType.StoredProcedure,
                CommandText = "NuevoCiudad"
            };
            cmd.Parameters.AddWithValue("@ID_CIUDAD", SqlDbType.Int).Value = IdCiudadCiudades;
            cmd.Parameters.AddWithValue("@NOMBRE_CIUDADES", SqlDbType.NVarChar).Value = NombreCiudadesCiudades;
            cmd.Parameters.AddWithValue("@ID_PROVINCIAS", SqlDbType.Int).Value = IdProvinciaCiudades;
            cmd.Parameters.AddWithValue("@ESTADO_CIUDADES", SqlDbType.Int).Value = EstadoCiudades;
            return cmd;
        }

        public SqlCommand ActualziarCiudad()
        {
            var cmd = new SqlCommand
            {
                CommandType = CommandType.Text,
                CommandText = "update CIUDADES set NOMBRE_CIUDADES = @NOMBRE_CIUDADES where ID_CIUDAD=@ID_CIUDAD;"
            };
            cmd.Parameters.AddWithValue("@ID_CIUDAD", SqlDbType.Int).Value = IdCiudadCiudades;
            cmd.Parameters.AddWithValue("@NOMBRE_CIUDADES", SqlDbType.NVarChar).Value = NombreCiudadesCiudades;
            return cmd;
        }
    }
}
